cid.js

```diff
@@ -145,11 +160,57 @@ export default multiformats => {
         this.version === other.version &&
         bytes.equals(this.multihash, other.multihash)
     }
+
+    get [Symbol.toStringTag] () {
+      return 'CID'
+    }
+
+    get [cidSymbol] () {
+      return true
+    }
+
+    /**
+     * Takes any input `value` and returns a `CID` instance if it was
+     * a `CID` otherwise returns `null`. If `value` is instanceof `CID`
+     * it will return value back. If `value` is not instance of this CID
+     * class, but is compatible CID it will return new instance of this
+     * `CID` class. Otherwise returs null.
+     *
+     * This allows two different incompatible versions of CID library to
+     * co-exist and interop as long as binary interface is compatible.
+     * @param {any} value
+     * @returns {CID|null}
+     */
+    static asCID (value) {
```

Comment:
Probably need some comments in here to detail exactly why all of this is the way that it is, it took me a while to realize this was correct.

Normally, an `instanceof` check would not be sufficient, because this class is created dynamically and you’d want this to succeed for *any* `CID` class generated by this code but bound to a different multiformats instance. 

However, this implementation is actually **ideal** in our case because we actually *want* to recast the CID bound to this specific multiformats implementation, so we want to fall back to the same handling we’d have below (which I believe is designed for easier interprocess communication).